### PR TITLE
Deprecate Azure Event Hubs PubSub component

### PR DESCRIPTION
**Pull request by berndverst**
Thank you for helping make the Dapr documentation better!

Deprecate Azure Event Hubs PubSub component

**Please follow this checklist before submitting:**

- [X] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [X] Commands include options for Linux, MacOS, and Windows within codetabs
- [X] New file and folder names are globally unique
- [X] Page references use shortcodes instead of markdown or URL links
- [X] Images use HTML style and have alternative text
- [X] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description
Deprecate Azure Event Hubs PubSub component

## Issue reference

https://github.com/dapr/components-contrib/issues/951

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md [hunk 1]
```diff
@@ -1,12 +1,16 @@
 ---
 type: docs
-title: "Azure Events Hub"
-linkTitle: "Azure Events Hub"
-description: "Detailed documentation on the Azure Event Hubs pubsub component"
+title: "Azure Events Hub (deprecated)"
```
Comment (greenie-msft) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md]: Suggestion to remove the "s". "Azure Events Hub" => "Azure Event Hub"

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
+linkTitle: "Azure Events Hub (deprecated)"
+description: "Detailed documentation on the Azure Event Hubs pubsub component (deprecated)"
 aliases:
   - "/operations/components/setup-pubsub/supported-pubsub/setup-azure-eventhubs/"
 ---
 
+{{% alert title="Deprecated" color="warning" %}}
+The Azure Event Hubs pubsub component is deprecated and has been removed from starting with release 1.4.
```
Comment (greenie-msft) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md]: "The Azure Event Hub pubsub component is deprecated and will no longer be available starting with release 1.4"

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md (hunk 1 above)
Comment (berndverst) [reply daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md]: Absolutely, I just appended "deprecated" to the misspelled name here.

Comment (berndverst) [reply daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md]:
```suggestion
title: "Azure Event Hubs (deprecated)"
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md (hunk 2 above)
Comment (berndverst) [reply daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md]:
```suggestion
The Azure Event Hub pubsub component is deprecated and will no longer be available starting with release 1.4.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
-description: "Detailed documentation on the Azure Event Hubs pubsub component"
+title: "Azure Event Hubs (deprecated)"
+linkTitle: "Azure Events Hub (deprecated)"
+description: "Detailed documentation on the Azure Event Hubs pubsub component (deprecated)"
 aliases:
   - "/operations/components/setup-pubsub/supported-pubsub/setup-azure-eventhubs/"
 ---
 
+{{% alert title="Deprecated" color="warning" %}}
+The Azure Event Hub pubsub component is deprecated and will no longer be available starting with release 1.4.
```
Comment (berndverst) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md]:
```suggestion
The Azure Event Hubs pubsub component is deprecated and will no longer be available starting with release 1.4.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md [hunk 4]
```diff
@@ -1,12 +1,16 @@
 ---
 type: docs
-title: "Azure Events Hub"
-linkTitle: "Azure Events Hub"
-description: "Detailed documentation on the Azure Event Hubs pubsub component"
+title: "Azure Event Hubs (deprecated)"
+linkTitle: "Azure Events Hub (deprecated)"
```
Comment (greenie-msft) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md]: Event Hubs

Comment (berndverst) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md]:
```suggestion
linkTitle: "Azure Event Hubs (deprecated)"
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/_index.md [hunk 5]
```diff
@@ -46,5 +46,5 @@ Table captions:
 
 | Name                                                      | Status | Component version | Since |
 |-----------------------------------------------------------|--------| ----------------| -- |
-| [Azure Events Hub]({{< ref setup-azure-eventhubs.md >}})  | Alpha  | v1 | 1.0 |
+| ~~[Azure Events Hub]({{< ref setup-azure-eventhubs.md >}})~~  | Deprecated  | ~~v1~~ | ~~1.0~~ |
```
Comment (berndverst) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/_index.md]:
```suggestion
| ~~[Azure Event Hubs]({{< ref setup-azure-eventhubs.md >}})~~  | Deprecated  | ~~v1~~ | ~~1.0~~ |
```

Comment (greenie-msft) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/_index.md]:
```suggestion
| ~~[Azure Event Hubs]({{< ref setup-azure-eventhubs.md >}})~~  | Deprecated  | ~~v1~~ | ~~1.0~~ |
```

**Review by greenie-msft — APPROVED** [review]

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md [hunk 6]
```diff
@@ -1,12 +1,16 @@
...
-description: "Detailed documentation on the Azure Event Hubs pubsub component"
+title: "Azure Event Hubs (deprecated)"
+linkTitle: "Azure Event Hubs (deprecated)"
+description: "Detailed documentation on the Azure Event Hubs pubsub component (deprecated)"
 aliases:
   - "/operations/components/setup-pubsub/supported-pubsub/setup-azure-eventhubs/"
 ---
 
+{{% alert title="Deprecated" color="warning" %}}
+The Azure Event Hubs pubsub component is deprecated and will no longer be available starting with release 1.4.
```
Comment (berndverst) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md]:
```suggestion
The Azure Event Hubs pubsub component is deprecated and will be removed in release 1.6. Please use the [Azure Event Hubs bindings]({{< ref "supported-bindings/eventhubs.md" >}}) instead.
```

**Comment by berndverst** [comment]
@greenie-msft @orizohar I updated the message here. Can you please take another look? We could merge this into 1.4 and then merge the same change into the current 1.3 via cherry pick to make sure people can see they shouldn't be using this component.

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md [hunk 7]
```diff
@@ -1,12 +1,16 @@
...
-description: "Detailed documentation on the Azure Event Hubs pubsub component"
+title: "Azure Event Hubs (deprecated)"
+linkTitle: "Azure Event Hubs (deprecated)"
+description: "Detailed documentation on the Azure Event Hubs pubsub component (deprecated)"
 aliases:
   - "/operations/components/setup-pubsub/supported-pubsub/setup-azure-eventhubs/"
 ---
 
+{{% alert title="Deprecated" color="warning" %}}
+The Azure Event Hubs pubsub component is deprecated and will be removed in release 1.6. Please use the [Azure Event Hubs bindings]({{< ref "supported-bindings/eventhubs.md" >}}) instead.
```
Comment (berndverst) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md]:
```suggestion
The Azure Event Hubs pubsub component is deprecated and will be removed in release 1.6. Please use the [Azure Event Hubs bindings]({{< ref "eventhubs.md" >}}) instead.
```

**Review by artursouza — APPROVED** [review]

**Review by AaronCrawfis — APPROVED** [review]

**Comment by artursouza** [comment]
Let's wait until our next conversation about this to decide on deprecation path.

**Review by artursouza — CHANGES_REQUESTED** [review]
Waiting to confirm on deprecation discussion.

**Review by orizohar — APPROVED** [review]

**Comment by greenie-msft** [comment]
@artursouza, we are still planning to start the deprecation process in 1.4?

**Comment by artursouza** [comment]
@greenie-msft Yes, we are. This PR can be merged.

**Comment by greenie-msft** [comment]
My understanding is that we are no longer deprecating this component. Instead we will re-write it. See here for details: https://github.com/dapr/components-contrib/issues/951

**Comment by yaron2** [comment]
> My understanding is that we are no longer deprecating this component. Instead we will re-write it. See here for details: [dapr/components-contrib#951](https://github.com/dapr/components-contrib/issues/951)

This PR can change to mention the current implementation should not be used. We will remove the warning once the issues are fixed.

**Comment by orizohar** [comment]
@artursouza based on the latest decision, sounds like we are not deprecating this component after all. Should we close this without merging?

**Comment by artursouza** [comment]
Not merging this anymore. We will rewrite the component instead.

**Comment by orizohar** [comment]
Closing without merging since the component will not be deprecated for now

**Pull request closed without merging**